src/com/seafile/seadroid2/SSLTrustManager.java

```diff
@@ -209,11 +208,12 @@ public void checkServerTrusted(X509Certificate[] chain, String authType)
                 return;
             }
 
+            List<X509Certificate> orderedChain = orderCerts(chain);
             try {
                 // First try to do default check
                 defaultTrustManager.checkServerTrusted(chain, authType);
+                customCheck(orderedChain, authType);
```

Comment:
The logic is not clear here. If the default certificate check is successful, why bother doing custom certificate check? What we need is:

- If the default certificate check passes, we need to verify the host name
- but if the custom check passes, the user has chosen to trust the certs before, so we don't need to verify host name again.